### PR TITLE
Documentation: Improve semantics in the block-based theme guide

### PR DESCRIPTION
**Pull request by amarinediary**
## Description
Changed "theme name" to "theme text domain". "Theme name" suggest the actual theme name with spaces and caps as defined in the style.css file example on line 66: `Theme Name: My first theme`.
In the given example, the "theme text domain" is actually used. Therefore to avoid any confusion it should be changed to "theme text domain", as per line 78: `Text Domain: myfirsttheme`.

## How has this been tested?
Purely a semantics update. For better understanding and consistency.

## Types of changes
Semantics update.

## Checklist:
Not applicable.

### CODE REVIEW

**Review by talldan — APPROVED** [review]
Thanks!

**Comment by talldan** [comment]
I've updated the PR title, as this is the text used in the plugin changelog.

**Comment by carolinan** [comment]
Note that since several plugin versions, the theme attribute is **not used** for template parts, this entire tutorial needs to be rewritten. It has not been rewritten yet because there are expected changes to theme.json.

**Comment by amarinediary** [comment]
@carolinan agreed. I think the documentation might play a huge role in people adopting full site editing. It is worth it to dumb it down and it will push people towards it. Including more graphical representations (screenshots and so on) might also be a more friendlier approach for upcoming guide.
